.github/workflows/go.yml

```diff
@@ -1,19 +1,39 @@
 name: go
 
 on:
-  - push
+  pull_request:
+
+permissions:
+  contents: read
+  pull-requests: write
+  security-events: write
+  actions: read
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Extract Go version
+        uses: arnested/go-version-action@8a203e9ff069cbbf4e3b65cb248101cfe307c71c # v1.1.18
+        id: go-version
+
+      - name: Split version code
+        uses: xom9ikk/split@10ba6c9f71c5953bc304e21781213e933b043891 #v1.1
+        id: split
+        with:
+          string: ${{ steps.go-version.outputs.go-mod-version }}
+          separator: .
+          limit: 2
```

Comment:
extract the go version from `go.mod` - so we don't need to update the action when we bump the go version.